### PR TITLE
BHBC-918: Edit General Information

### PR DESCRIPTION
**Pull request by NickPhura**
<details open>
  <summary><a href="https://quartech.atlassian.net/browse/BHBC-918" title="BHBC-918" target="_blank">BHBC-918</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
      <td>Edit General Information section in the View Project page </td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Story
        Story
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>In Progress</td>
    </tr>
    <tr>
      <th>Points</th>
      <td>N/A</td>
    </tr>
    <tr>
      <th>Labels</th>
      <td>-</td>
    </tr>
  </table>
</details>

---

# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-918

## This PR contains the following changes

- General information edit
- Fixed a bug in the APIError class (wasn't parsing the right fields from the error response object)
  - The error messages thrown in the API now properly show up in the error dialog on the front-end 
  - Updated some of the backend error messages to read a bit nicer

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [x] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [x] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [x] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [x] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: api/src/queries/project/project-update-queries.test.ts [hunk 1]
```diff
@@ -49,6 +52,135 @@ describe('getCoordinatorByProjectSQL', () => {
...
+        revision_count: 1
+      }),
+      null,
+      1
+    );
+
+    expect(response).to.not.be.null;
+  });
+
+  it('returns valid sql when only objectives data is provided', () => {
```
Comment (sdevalapurkar) [inline api/src/queries/project/project-update-queries.test.ts]:
```suggestion
  it('returns valid sql when only coordinator data is provided', () => {
```

File: app/src/components/dialog/YesNoDialog.test.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        onClose={handleOnClose}
+        onNo={handleOnNo}
+        onYes={handleOnYes}
+      />
+    </div>
+  );
+};
+
+describe('EditDialog', () => {
+  it('matches the snapshot when not open, with no error message', () => {
```
Comment (sdevalapurkar) [inline app/src/components/dialog/YesNoDialog.test.tsx]:
```suggestion
  it('matches the snapshot when not open', () => {
```

File: app/src/features/projects/view/components/GeneralInformation.test.tsx [hunk 3]
```diff
@@ -51,7 +87,44 @@ describe('GeneralInformation', () => {
...
+          end_date: '2020-05-20',
+          revision_count: 2
+        }
+      });
+
+      expect(mockRefresh).toBeCalledTimes(1);
+    });
+  });
+
+  it('displays an error dialog when fetching the update data fails', async () => {
```
Comment (sdevalapurkar) [inline app/src/features/projects/view/components/GeneralInformation.test.tsx:107]: can we add the tests that I added yesterday to this component as well so it tests the `onClose` callback as well and the apierror stuff

Comment (NickPhura) [reply app/src/features/projects/view/components/GeneralInformation.test.tsx:107]: yup!

**Review by sdevalapurkar — DISMISSED** [review]
awesome!!!

**Comment by sdevalapurkar** [comment]
almost 53% woohoo @NickPhura !!! 🥇

**Review by anissa-agahchen — APPROVED** [review]
Looks great!

**Review by sdevalapurkar — APPROVED** [review]

**Pull request merged**